lib/lambda_ethereum_consensus/validator/validator.ex

```diff
@@ -31,35 +32,27 @@ defmodule LambdaEthereumConsensus.Validator do
 
   @default_graffiti_message "Lambda, so gentle, so good"
 
-  @type validator :: %{
-          index: non_neg_integer() | nil,
-          pubkey: Bls.pubkey(),
-          privkey: Bls.privkey()
-        }
-
   # TODO: Slot and Root are redundant, we should also have the duties separated and calculated
   # just at the begining of every epoch, and then just update them as needed.
   @type state :: %__MODULE__{
           slot: Types.slot(),
           epoch: Types.epoch(),
           root: Types.root(),
           duties: Duties.duties(),
-          validator: validator(),
+          index: non_neg_integer() | nil,
+          keystore: Keystore.t(),
```

Comment:
The Validator state is weird but this will be a lot better in the coming days. Lot of this is going to change in the refactor I'm working on, I already removed slot, epoch, root and even duties from here, it should be easier by then hopefully.

Reply:
I agree. I just hope that this doesn't add too much noise to your PR!